JarrydWorland/2021_Capstone_GP2_Team
# Refactor player movement & animations

## Changes include
- Added player sprites (778cf16)
    - Reorganized folder structure.
- Added player animations (a200c54)
    - Unified animations to use a single animation controller.
- Added player scripts (dbda9ba)
    - Added `PlayerMovementBehaviour` for movement logic
    - Placed animator specific logic within `PlayerAnimatorBehaviour`. This might not be possible for non-player animations, and so for the sake of consistency throughout the project this might not be a permanent change.
- Added player prefab (2e9ea87)
- Removed gravity from project physics settings (1ba5496)

## Prerequisites
- [refactor-input-management](https://github.com/JarrydWorland/2021_Capstone_GP2_Team/tree/refactor-input-management) although no PR exists for it as of writing this.

## How to test
Place the player prefab into the scene and walk around, ensure the animations look correct. Currently shooting has not been re-implemented so to test shooting animations, open the `PlayerAnimationController` and enable the shooting layer by setting the layer weight. Temporarily for shooting animations the player will always aim towards the scene origin.
